docs/advanced/virtual-threads.md

```diff
@@ -0,0 +1,74 @@
+## Datafetching threading model without Virtual Threads
+
+Without virtual threads, the DGS framework does not create or manage threads for data fetchers.
+Every data fetcher will run on the same request thread.
+This effectively means that data fetchers do not have any concurrent/parallel behavior, except when explicitly returning `CompletableFuture`.
+The reason for not having concurrent behavior by default is that this would require using a thread per field, which would not scale well.
+
+You can achieve concurrent data fetchers returning `CompletableFuture` from a data fetcher.
```

Comment:
Consider rephrasing to "To achieve concurrency in data fetchers, you can return `CompletableFuture` from a data fetcher.